### PR TITLE
contents of

### LINKED ISSUE #50: StateChanges.java:70-72: Move to separate class Create...
The puzzle `37-abe81894` from #37 has to be resolved: 

https://github.com/eo-cqrs/cmig/blob/bca9d14b0714d939d2bd40aec62747583469aaf4/src/main/java/io/github/eocqrs/cmig/sha/StateChanges.java#L70-L72

The puzzle was created by @h1alexbel on 16-Aug-23. 

Estimate: 30 minutes,  role: DEV. 

If you have any technical questions, don't ask me, submit new tickets instead. The task will be \"done\" when the problem is fixed and the text of the puzzle is _removed_ from the source code. Here is more about [PDD](http://www.yegor256.com/2009/03/04/pdd.html) and [about me](http://www.yegor256.com/2017/04/05/pdd-in-action.html).

### PR DESCRIPTION
**Pull request by h1alexbel**
@l3r8yJ take a look, please
closes #50

---

### Detailed summary
- The PR introduces a new class `ContentsOf` that reads the contents of a file.
- The `StateChanges` class now uses `ContentsOf` to retrieve the contents of files in a directory.
- A new test class `ContentsOfTest` is added to test the `ContentsOf` class.

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by zoeself** [comment]
@h1alexbel thank you for your Pull Request. I'll assign someone to review it soon. 

 If this PR solves a ``todo`` from the code, please don't forget to remove it.

**Comment by zoeself** [comment]
@h1alexbel please review this Pull Request. Deadline (when it should be merged or closed) is ``2023-08-19T08:39:05.835334``.

You should check if the requirements have been implemented (partially or in full), if there are unit tests covering the changes and if the CI build passes. Feel free to reject the PR or ask for changes if it's too big or not clear enough.

Estimation here is ``30 minutes``, that's how much you will be paid. You will be paid even if this PR gets rejected.

**Comment by h1alexbel** [comment]
@zoeself deregister

**Comment by zoeself** [comment]
> @zoeself deregister

@h1alexbel ok, I've removed this task from scope. I'm not managing it anymore.

**Comment by h1alexbel** [comment]
@zoeself register

**Comment by zoeself** [comment]
> @zoeself register

@h1alexbel I've just registered this ticket as a task and will assign it to someone soon. Thanks!

**Comment by zoeself** [comment]
@l3r8yJ please review this Pull Request. Deadline (when it should be merged or closed) is ``2023-08-19T08:49:05.379595``.

You should check if the requirements have been implemented (partially or in full), if there are unit tests covering the changes and if the CI build passes. Feel free to reject the PR or ask for changes if it's too big or not clear enough.

Estimation here is ``30 minutes``, that's how much you will be paid. You will be paid even if this PR gets rejected.

File: src/main/java/io/github/eocqrs/cmig/sha/ContentsOf.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    final String cmg,
+    final String name
+  ) {
+    this.cmig = cmg;
+    this.file = name;
+  }
+
+  @Override
+  public String asString() throws Exception {
+    return new UnixizedOf(
```
Comment (l3r8yJ) [inline src/main/java/io/github/eocqrs/cmig/sha/ContentsOf.java]: i think this is possible here

```suggestion
  @SneakyThrows
  @Override
  public String asString() {
    return new UnixizedOf(
```

**Review by l3r8yJ — CHANGES_REQUESTED** [review]
@h1alexbel take a look at my comments below, please

**Comment by h1alexbel** [comment]
@l3r8yJ take a look again, please

**Review by l3r8yJ — APPROVED** [review]
@h1alexbel looks good to me!

**Comment by h1alexbel** [comment]
@rultor merge

**Comment by zoeself** [comment]
@l3r8yJ thank you for resolving this ticket. I've just added it to your active invoice. You can always check all your invoices and more on the [Contributor Dashboard](https://self-xdsd.com).

**Pull request merged**